tests/cpp/metric/test_multiclass_metric.cc

```diff
@@ -4,17 +4,12 @@
 
 #include "../helpers.h"
 
-using Arg = std::pair<std::string, std::string>;
-
-#if defined(__CUDACC__)
-#define N_GPU() Arg{"n_gpus", "1"}
-#else
-#define N_GPU() Arg{"n_gpus", "0"}
-#endif
-
-inline void TestMultiClassError(std::vector<Arg> args) {
-  xgboost::Metric * metric = xgboost::Metric::Create("merror");
-  metric->Configure(args);
+inline void TestMultiClassError(xgboost::GPUSet devices) {
```

Comment:
```suggestion
inline void TestMultiClassError(xgboost::GPUSet const& devices) {
```
to be consistent with uses of `GPUSet` in the `Learner` class.